### DISCUSSION TITLE
EarlyStopping patience 100 exceeded.   Anyone know what causes this?

### DISCUSSION THREAD

**Discussion by aldrichg9** [ultralytics/yolov5#5204] [Q&A] [answered]
I always assumed that a low mAP would cause "EarlyStopping patience 100 exceeded, stopping training."
But take a look at this below screenshot.  mAP was looking great.
Does anyone know what scenarios will stop the training this way?

![image](https://user-images.githubusercontent.com/25693009/137484433-150f4b24-b1af-469b-b26a-845d88f007db.png)

**Comment by Matias2379** [comment, marked as answer]
Hi @aldrichg9 ,  early stopping is used to avoid overfitting. The "patience" parameter tells how many epochs the model will continue training after the val los stops improving against train loss. In this case, after 100 epochs of patience, the model stops training and (usually) takes the weights from the epoch (NOW - 100). 
I hope it helps.
Regards, Matias

**Comment by glenn-jocher** [comment]
@aldrichg9 this basically means your results have not improved for 100 epochs, so training is stopping under the assumption you're overfitting and will not improve anymore going forward.

You can modify the patience parameter yourself:
```
python train.py --patience 30
```

**Comment by aldrichg9** [comment]
Thank you for the replies.
